pkg/flags/environment.go

```diff
@@ -0,0 +1,71 @@
+package flags
+
+import "github.com/spf13/pflag"
+
+type EnvironmentalFlags struct {
+	Platform     string
+	Network      string
+	Upgrade      string
+	Topology     string
+	Architecture string
+	Installer    string
+	Config       []string
+	Facts        map[string]string
+	Version      string
+}
+
+func NewEnvironmentalFlags() *EnvironmentalFlags {
+	return &EnvironmentalFlags{}
+}
+
+func (f *EnvironmentalFlags) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&f.Platform,
+		"platform",
+		"",
+		"The hardware or cloud platform (\"aws\", \"gcp\", \"metal\", ...). Since: v1.0")
```

Comment:
After looking at https://issues.redhat.com/browse/TRT-1853, I don't think that this versioning will be good enough for our purposes. This is human readable, but we essentially need to have a mapping of flags per version. Maybe the OTE can include an endpoint that lists all available environment flags?

Reply:
Two purposes to the version:
- Historical knowledge when a flag was added (your solution addresses that)
- origin needs to know which flags to send to which versions of the binaries.  It would be nice to have something like a map it could just look at (vendored or just copied over, not sure if we want to vendor OTE yet in origin)

So maybe something simple like this:

```go
// FlagVersions holds the "Since" version metadata for each flag.
var FlagVersions = map[string]string{
	"platform":     "v1.0",
	"network":      "v1.0",
	"upgrade":      "v1.0",
	"topology":     "v1.0",
	"architecture": "v1.0",
	"installer":    "v1.0",
	"config":       "v1.0",
	"fact":         "v1.0",
	"version":      "v1.0",
	"new-feature":  "v1.1", // Example of a new flag added in v1.1
}
```

Reply:
(It might also be nice to have something that checks to make sure all BindFlags have an entry in the map and if not panics)

Reply:
Perfect, that is essentially what I am implementing right now. My idea was to return a json list with that info with a new command on the binary called `env-flags`. So origin would use the new command to get a list of flags available in that binary prior to setting the appropriate ones.

Reply:
It's probably ok for origin just to base it off the version that the `info` command returns

Reply:
The actual version doesn't need to be passed anywhere as the command will always just return the flags applicable to its current version. I have pushed the changes now, hopefully it makes sense

Reply:
Alternatively, we could have the available flags returned as part of the `info` command, if that makes more sense.

Reply:
The info command already returns the version, I think I prefer that over returning the list of flags supported.  The code in origin will be simpler.

Reply:
So the mapping of flags to version will exist in origin only then, right?